runtime/interpreter/interpreter_expression.go

```diff
@@ -212,6 +221,23 @@ func (interpreter *Interpreter) memberExpressionGetterSetter(memberExpression *a
 				}
 			}
 
+			// Return a reference, if the member is accessed via a reference.
+			//
+			// However, for attachments, `self` is always a reference.
+			// But we do not want to return a reference for `self.something`.
+			// Otherwise, things like `destroy self.something` would become invalid.
+			// Hence, special case `self`, and return a reference only if the member is not accessed via self.
```

Comment:
Do we only need to handle the case where there's a member expression, and the LHS is `self`, e.g. `self.foo`? Do we also need to handle nested cases, e.g. `self.foo.bar`?

Reply:
Because the expression visitor is recursive, handling only one level is sufficient. i.e. in `self.foo.bar`, the result of `.bar` depends on the result of `self.foo`. So as long as evaluating `self.foo` returns the correct result, then `.bar` doesn't need to be special-cased.

Added a test: eca92dfc94ed739876e321a3b80464398c8a45db